### ISSUE TITLE
UnorderedFoldable#isEmpty default implementation is incorrect

### ISSUE THREAD

**Issue by durban** [typelevel/cats#2584]
The default implementation of `UnorderedFoldable#isEmpty` (and thus `nonEmpty`) seems to be incorrect.

```scala
val inst: cats.UnorderedFoldable[Set] = new cats.UnorderedFoldable[Set] {
  def unorderedFoldMap[A, B](s: Set[A])(f: A => B)(implicit B: cats.kernel.CommutativeMonoid[B]): B =
    cats.instances.set.catsStdInstancesForSet.unorderedFoldMap(s)(f)
}

inst.isEmpty(Set.empty) // returns false
```

**Comment by ceedubs** [comment]
Ouch. It looks like the implementations of `isEmpty` and `nonEmpty` are switched around. The fix is easy, but we should also add some tests for this. This is a super busy week for me. Does anybody want to give it a go?

**Comment by barambani** [comment]
I can help with this if that's ok

**Comment by kailuowang** [comment]
closed by #2586
